assembl/static2/js/app/components/debate/survey/post.jsx

```diff
@@ -110,63 +130,107 @@ class Post extends React.Component {
+    if (!post.publicationState || post.publicationState.startsWith('DELETED')) {
+      return null;
+    }
+
+    const { contentLocale, lang, screenWidth, originalLocale, questionId } = this.props;
     const { debateData } = this.props.debate;
     const { bodyEntries } = post;
     const translate = contentLocale !== originalLocale;
 
-    let body;
-    if (bodyEntries.length > 1) {
-      // first entry is the translated version, example localeCode "fr-x-mtfrom-en"
-      // second entry is the original, example localeCode "en"
-      body = translate ? bodyEntries[0].value : bodyEntries[1].value;
-    } else {
-      // translation is not enabled or the message is already in the desired locale
-      body = bodyEntries[0].value;
+    // to update the question header when we delete the post or add/remove a sentiment
+    const updateQuestionQuery = {
+      query: QuestionQuery,
+      variables: {
+        id: questionId,
+        lang: lang
+      }
+    };
+
+    let body = '';
+    if (bodyEntries) {
+      if (bodyEntries.length > 1) {
+        // first entry is the translated version, example localeCode "fr-x-mtfrom-en"
+        // second entry is the original, example localeCode "en"
+        body = translate ? bodyEntries[0] && bodyEntries[0].value : bodyEntries[1] && bodyEntries[1].value;
+      } else {
+        // translation is not enabled or the message is already in the desired locale
+        body = bodyEntries[0] ? bodyEntries[0].value : '';
+      }
     }
+
+    const sentimentCounts = post.sentimentCounts;
+    const currentCounts = {
+      like: sentimentCounts && sentimentCounts.like ? sentimentCounts.like : 0,
+      disagree: sentimentCounts && sentimentCounts.disagree ? sentimentCounts.disagree : 0
+    };
     const likeComponent = (
       <div
         className={post.mySentiment === 'LIKE' ? 'sentiment sentiment-active' : 'sentiment'}
         onClick={(event) => {
-          this.handleSentiment(event, 'LIKE');
+          this.handleSentiment(event, 'LIKE', [updateQuestionQuery], currentCounts);
         }}
       >
         <Like size={25} />
       </div>
     );
+
     const disagreeComponent = (
       <div
         className={post.mySentiment === 'DISAGREE' ? 'sentiment sentiment-active' : 'sentiment'}
         onClick={(event) => {
-          this.handleSentiment(event, 'DISAGREE');
+          this.handleSentiment(event, 'DISAGREE', [updateQuestionQuery], currentCounts);
         }}
       >
         <Disagree size={25} />
       </div>
     );
 
-    const { displayName, isDeleted } = post.creator;
+    let creatorName = '';
+    let userCanDeleteThisMessage = false;
+    if (post.creator) {
+      const { displayName, isDeleted } = post.creator;
+      const connectedUserId = getConnectedUserId();
+      userCanDeleteThisMessage =
+        (post.creator && (connectedUserId === String(post.creator.userId) && connectedUserCan(Permissions.DELETE_MY_POST))) ||
```

Comment:
brackets around please ```connectedUserCan(Permissions.DELETE_MY_POST))) || connectedUserCan(Permissions.DELETE_POST);```

Reply:
Also, not that it makes much of a difference, but why not do an ```int(connectedUserId === post.creator.userId``` 

check instead?

Reply:
about the brackets, we don't want parens around the 2 permission checks, the second one checks that user can delete the message even if he has not created it

Reply:
about the cast, in fact I took the code from `postActions.jsx`. I guess that the only difference would be that if `connectedUserId` is `undefined`, `Number(connectedUserId)` would return `NaN` (I don't know if it can happen)